sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java

```diff
@@ -58,10 +58,16 @@
 
   /**
    * The column IDs of the physical ORC file schema which are required by this reader.
-   * -1 means this required column doesn't exist in the ORC file.
+   * -1 means this required column is partition column, or it doesn't exist in the ORC file.
    */
   private int[] requestedColIds;
 
+  /**
+   * The column IDs of the ORC file partition schema which are required by this reader.
```

Comment:
it's not column ID, it's the index of the partition column

Reply:
according to https://github.com/apache/spark/pull/23387/files#diff-8e7e05590803a656e46183bc94642ab8R211